src/libraries/System.Text.Json/gen/Model/TypeGenerationSpec.cs

```diff
@@ -104,8 +109,7 @@ public bool IsFastPathSupported()
                     foreach (PropertyGenerationSpec property in PropertyGenSpecs)
                     {
                         if (property.PropertyType.SpecialType is SpecialType.System_Object ||
-                            property.NumberHandling is JsonNumberHandling.AllowNamedFloatingPointLiterals
-                                                    or JsonNumberHandling.WriteAsString ||
+                            property.NumberHandling != null ||
```

Comment:
Should the same check for type infos apply here?

Reply:
It's causing test failures for cases where you have properties containing POCOs annotated with `JsonNumberHandling.Strict` and you have global configuration specifying `WriteAsString`. The fast-path method cannot pass the property-level configuration to downlevel `JsonSerializer.Serialize` calls so it ends up picking up the global configuration incorrectly. Probably not a super common scenario but I figured we should be conservative for now.